internal/service/file/file_folder.go

```
@@ -489,20 +489,34 @@ func (s *FileService) MoveFiles(ctx context.Context, uid string, srcFileIDs []st
 		}
 	}
 
 	return true, ""
 }
 
+// renameLinkedDocuments renames every knowledgebase document linked to the
+// given file so the new name is reflected in all linked datasets.
+func (s *FileService) renameLinkedDocuments(ctx context.Context, fileID, newName string) error {
+	informs, err := s.file2DocumentDAO.GetByFileID(ctx, dao.DB, fileID)
+	if err != nil {
+		return nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 504) but it returns nil

(nilerr)